### DISCUSSION TITLE
Using InstanceSplitter transformation with existing models.

### DISCUSSION THREAD

**Discussion by nietoo** [awslabs/gluonts#1818] [Q&A] [answered]
Hello.

Is there a way to use existing models with an InstanceSplitter transformation? I am trying to do the following (as an example):

```
transformation = create_transformation('1D', DATA_PROCESSING.CONTEXT_LENGTH, DATA_PROCESSING.PREDICTION_LENGTH)

train_tf = transformation(iter(train_ds), is_train=True)
test_tf = transformation(iter(test_ds), is_train=False)

train_tds = TransformedDataset(train_ds, transformation)
test_tds = TransformedDataset(test_ds, transformation)

estimator = SimpleFeedForwardEstimator(
    num_hidden_dimensions=FEED_FORWARD.NUM_DIMENSIONS,
    prediction_length=DATA_PROCESSING.PREDICTION_LENGTH,
    context_length=DATA_PROCESSING.CONTEXT_LENGTH,
    freq='1D',
    trainer=Trainer(
        ctx="cpu",
        epochs=5,
        learning_rate=1e-3,
        hybridize=False,
        num_batches_per_epoch=100
    )
)

predictor = estimator.train(train_tds)
```
(where create_transformation returns a Chain with InstanceSplitter as the last transformation)

However, I am getting a `KeyError: 'target'` error since the name of target field is changed by the InstanceSplitter. Is there any way to use the transformed dataset with an existing model, or do I need to create a custom estimator inheriting an existing one?

Thanks in advance.

**Comment by mbohlkeschneider** [comment, marked as answer]
Hi @nietoo,

thank you for raising your issue. With the existing estimators there is usually no need to do the transformation yourself because it would be already implemented in the `Estimator` class. For specific advice, I would kindly ask you to post a minimal working example.

**Reply by nietoo** [reply]
Thank you, that makes sense. In hindsight, I should've probably looked at the implementation of the estimator before asking.
